backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/outputGenerators/DocxUtils.java

```diff
@@ -275,5 +112,45 @@ private static void replaceParagraph(XWPFParagraph paragraph, String placeHolder
             }
         }
     }
-}
 
+    public static void addBulletList(XWPFDocument document, List<String> bulletListItems, boolean bulletListItemsAsLink) throws XmlException {
+        CTNumbering cTNumbering = CTNumbering.Factory.parse(cTAbstractNumBulletXML);
+        CTAbstractNum cTAbstractNum = cTNumbering.getAbstractNumArray(0);
+        XWPFAbstractNum abstractNum = new XWPFAbstractNum(cTAbstractNum);
+        XWPFNumbering numbering = document.createNumbering();
+        BigInteger abstractNumID = numbering.addAbstractNum(abstractNum);
+        BigInteger numID = numbering.addNum(abstractNumID);
+
+        for (int i = 0; i < bulletListItems.size(); i++) {
+            String bulletItem = bulletListItems.get(i);
+            XWPFParagraph paragraph = document.createParagraph();
+            paragraph.setNumID(numID);
+            if (bulletListItemsAsLink) {
+                addHyperLink(paragraph, "bookmark_" + bulletItem, bulletItem);
+            } else {
+                setText(paragraph.createRun(), bulletItem);
+            }
+            if (i < bulletListItems.size() - 1) {
+                paragraph.setSpacingAfter(0);
+            }
+        }
+    }
+
+    public static void addHyperLink(XWPFParagraph paragraph, String hyperlinkAnchor, String hyperlinkText) {
+        CTHyperlink cLink = paragraph.getCTP().addNewHyperlink();
+        cLink.setAnchor(hyperlinkAnchor.replaceAll(" ", "_"));
```

Comment:
Is that able to handle all whitespace, e.g. tabs, non-breaking-whitespace and newlines?

Reply:
It should be all whitespaces replaced (replaceAll("\\s+", "")).